core/src/listener/api.ts

```diff
@@ -58,3 +59,94 @@ export async function getListener({
     throw new OraklError(OraklErrorCode.GetListenerRequestFailed)
   }
 }
+
+/**
+ * @param {string} service
+ * @return {Promise<IBlock>}
+ * @exception {FailedToGetObservedBlock}
+ */
+export async function getObservedBlock({ service }: { service: string }): Promise<IBlock> {
+  try {
+    const endpoint = buildUrl(ORAKL_NETWORK_API_URL, `blocks/observed?service=${service}`)
+    return (await axios.get(endpoint))?.data
+  } catch (e) {
+    throw new OraklError(OraklErrorCode.FailedToGetObservedBlock)
+  }
+}
+
+/**
+ * @param {string} service
+ * @return {Promise<IBlocks[]>}
+ * @exception {FailedToGetUnprocessedBlocks}
+ */
+export async function getUnprocessedBlocks({ service }: { service: string }): Promise<IBlock[]> {
+  try {
+    const endpoint = buildUrl(ORAKL_NETWORK_API_URL, `blocks/unprocessed?service=${service}`)
+    return (await axios.get(endpoint))?.data
+  } catch (e) {
+    throw new OraklError(OraklErrorCode.FailedToGetUnprocessedBlocks)
+  }
+}
+
+/**
+ * @param {string} service
+ * @param {number} blockNumber
+ * @return {Promise<void>}
+ * @exception {FailedInsertUnprocessedBlock}
+ */
+export async function insertUnprocessedBlocks({
+  blocks,
+  service
+}: {
+  service: string
+  blocks: number[]
+}): Promise<void> {
+  try {
+    const endpoint = buildUrl(ORAKL_NETWORK_API_URL, `blocks/unprocessed`)
+    return (await axios.post(endpoint, { service, blocks }))?.data
+  } catch (e) {
+    throw new OraklError(OraklErrorCode.FailedInsertUnprocessedBlock)
+  }
+}
+
+/**
+ * @param {string} service
+ * @param {number} blockNumber
+ * @return {Promise<IBlock>}
+ * @exception {FailedDeleteUnprocessedBlock}
+ */
+export async function deleteUnprocessedBlock({
+  blockNumber,
+  service
+}: {
+  blockNumber: number
+  service: string
+}): Promise<IBlock> {
+  try {
+    const endpoint = buildUrl(ORAKL_NETWORK_API_URL, `blocks/unprocessed/${service}/${blockNumber}`)
+    return (await axios.delete(endpoint))?.data
+  } catch (e) {
+    throw new OraklError(OraklErrorCode.FailedDeleteUnprocessedBlock)
+  }
+}
+
+/**
+ * @param {string} service
+ * @param {number} blockNumber
+ * @return {Promise<IBlock>}
+ * @exception {FailedUpsertObservedBlock}
+ */
+export async function upsertObservedBlock({
```

Comment:
@Intizar-T There has to be check whether we do not upsert block number with smaller value. If it can be done through `INSERT` query, that would be the best.

Reply:
LGTM!